Runtimes/Readme.md

```diff
@@ -0,0 +1,99 @@
+# Swift Runtime Libraries
+
+This directory contains the pieces of the Swift runtime libraries.
+
+## Development
+
+While we're bringing up the new standard library build, please do not commit the
+standard library source files. Use the `Resync.cmake` file to copy the files as
+needed.
+
+```sh
+$ cmake -P Resync.cmake
+```
+
+> [!IMPORTANT]
+> Re-run this script after updating your Swift checkout to ensure that you are
+> building the latest standard library sources.
+
+Once the migration is completed, we will be deleting this script. It
+is a temporary workaround to avoid trying to keep multiple sets of files in
+sync.
+
+## Layering
+
+```
+╔═══════════════╗
+║               ║
+║    Testing    ║
+║               ║
+╠───────────────╣
+│               │
+│ Supplemental  │
+│               │
+├───────────────┤
+│               │
+│    Overlay    │
+│               │
+├───────────────┤
+│               │
+│     Core      │
+│               │
+└───────────────┘
+```
+
+### Core
+
+The _Core_ project contains the basic datatypes and underpinnings used by the
+rest of the libraries that make up the standard library. The _Core_ libraries
+must be built first.
+The _Core_ project provides the following libraries:
+ - `swiftCore`
+ - `swift_Concurrency`
```

Comment:
What makes the concurrency library "core"? Is it because it's implicitly imported by all Swift modules? If so, then `StringProcessing` is also core in that regard. I also envision that `Synchronization` will be depended upon by the concurrency library sooner or later.

Reply:
The only thing that's truly core IMO is `libswiftCore` 😄

Reply:
Generally things for which there is syntactic sugar that implicitly pulls in the module should be a "core" library, but without creating a cyclic dependency between layers.

More specifically:
 - Things that crash the compiler if the modules don't exist is `Core` (though we should probably make the compiler a bit more resilient to missing modules too and just say, "this runtime doesn't have X support")
 - Things that do not depend on the platform overlay can be `Core` (platform overlay can't depend on things that depend on the platform overlay, which currently rules out Synchronization and StringProcessing).
 - Things that are required by platform overlays must be in `Core` for the overlay to depend on it.

IIRC, at least the Darwin overlay contains MainActor annotations, requiring the concurrency runtime.

Reply:
Okay, the overlay doesn't seem to have MainActor annotations, but folks have expressed interest in using concurrency in their platform overlays.

Reply:
> Things that crash the compiler if the modules don't exist is Core (though we should probably make the compiler a bit more resilient to missing modules too and just say, "this runtime doesn't have X support")

So... StringProcessing?

```swift
// '-disable-implicit-string-processing-import' to simulate not having the module

let x = /\w+/
```

Reply:
That doesn't seem to compile... ?

```
ewilde@Pancake /tmp %  cat test.swift 
let x = /\w+/
ewilde@Pancake /tmp %  swiftc test.swift 
test.swift:1:9: error: '/' is not a prefix unary operator
1 | let x = /\w+/
  |         `- error: '/' is not a prefix unary operator
2 | 

test.swift:1:10: error: cannot find operator '+/' in scope
1 | let x = /\w+/
  |          `- error: cannot find operator '+/' in scope
2 | 

test.swift:1:10: error: cannot find 'w' in scope
1 | let x = /\w+/
  |          `- error: cannot find 'w' in scope
2 | 

test.swift:1:10: error: invalid component of Swift key path
1 | let x = /\w+/
  |          `- error: invalid component of Swift key path
2 | 
ewilde@Pancake /tmp %  swiftc --version
swift-driver version: 1.113 Apple Swift version 6.0 (swiftlang-6.0.0.7.6 clang-1600.0.24.1)
Target: arm64-apple-macosx15.0
```

It looks like the overlay import is just the variadics generator, so we could probably sink the library lower. May I ask what you're hoping to get by sinking it down into `Core` though? Are platform overlays trying to use regex?

Reply:
`-swift-version 6`

Reply:
Still not a compiler crash :)

```
ewilde@Pancake /tmp %  swiftc -Xfrontend -disable-implicit-string-processing-module-import test.swift -swift-version 6
test.swift:1:9: error: missing 'Regex' declaration, probably because the '_StringProcessing' module was not imported properly
1 | let x = /\w+/
  |         `- error: missing 'Regex' declaration, probably because the '_StringProcessing' module was not imported properly
2 | 
```

But anyway, there's project-level parallelism in there that we could take advantage of if they are kept separate. 
We really only need to drive it down into `Core` if folks are trying/wanting to use it in the platform overlay itself. Otherwise, it doesn't matter if it's built in the supplemental libraries or not.